cmd/bbolt/main.go

```diff
@@ -1166,9 +1166,7 @@ func (cmd *benchCommand) runWrites(db *bolt.DB, options *BenchOptions, results *
 		cmd.startProfiling(options)
 	}
 
-	finishChan := make(chan interface{})
-	go checkProgress(results, finishChan, cmd.Stderr)
-	defer close(finishChan)
+	defer results.StartProgress(cmd.Stderr).StopProgress()
```

Comment:
I tend to keep the previous implementation, which has better readability to me. It also looks weird to bind methods `StartProgress` and `StopProgress` with receiver `results *BenchResults`; the methods do not actually start or stop the read/write process, instead they create the monitoring goroutine.

Reply:
It doesn't start/stop read/write "process", it starts/stops "progress". I can rename it as `StartMonitoring`/`StopMonitoring` if it's confusing.

Chaining method calls in defer is pretty common pattern actually. One example from [stdlib](https://pkg.go.dev/runtime/trace#StartRegion):
> defer trace.StartRegion(ctx, "myTracedRegion").End()

If you're still not okay with this, I can revert the whole change but only keep `sync/atomic` changes that fixes the bug.

Reply:
> Chaining method calls in defer is pretty common pattern actually. One example from stdlib:

Thanks for the clarification. It seems like a separate enhancement. You enhanced the progress goroutine,  not only print progress info periodically, but also on demand. Can we review this in a separate PR?

> only keep sync/atomic changes that fixes the bug.

That would be a straightforward change, and should be approved & merged soon.